GitCommands/Remotes/GitRemoteManager.cs

```diff
@@ -274,8 +274,26 @@ public GitRemoteSaveResult SaveRemote(GitRemote remote, string remoteName, strin
             bool remoteDisabled = false;
             if (creatingNew)
             {
-                output = module.AddRemote(remoteName, remoteUrl);
-                updateRemoteRequired = true;
+                if (GetEnabledRemoteNames().FirstOrDefault(r => r == remoteName) != null)
+                {
+                    return new GitRemoteSaveResult($"An active remote named \"{remoteName}\" already exists.", false);
+                }
+                else if (GetDisabledRemoteNames().FirstOrDefault(r => r == remoteName) != null)
+                {
+                    return new GitRemoteSaveResult($"An inactive remote named \"{remoteName}\" already exists.", false);
+                }
+                else
+                {
+                    output = module.AddRemote(remoteName, remoteUrl);
+
+                    // If output was returned, something went wrong
+                    if (output.Length > 0)
```

Comment:
If Module.AddRemote succeeds, it returns an empty string; so if it fails, we shouldn't fall through as it did before. This was already broken. You can test it by adding a new remote, but not populating the URL field and clicking Save Changes.

Reply:
Nice!